src/runtime/README.md

```diff
@@ -86,6 +86,26 @@ following locations (in order):
 
 > **Note:** For both binaries, the first path that exists will be used.
 
+#### Drop-in configuration file fragments
+
+To enable changing configuration without changing the configuration file
+itself, drop-in configuration file fragments are supported.  Once a
+configuration file is parsed, if there is a subdirectory called `config.d` in
+the same directory as the configuration file its contents will be enumerated
+in alphabetical order and each item will be parsed as a config file.  Settings
+loaded from these configuration file fragments override settings loaded from
+the main configuration file and earlier fragments.  Users are encouraged to use
+familiar naming conventions to order the fragments (e.g. `config.d/10-this`,
+`config.d/20-that` etc.).
+
+Non-existent or empty `config.d` directory is not an error (in other words,
+not using configuration file fragments is fine).  On the other hand, if
+fragments are used, they must be valid - any errors while parsing fragments
+are treated the same as errors while parsing the main configuration file.  A
```

Comment:
Yes, I can add some (malformed TOML or failure to open/read the files could be some of the more common ones and worth a mention).

The whole point here though is that the error handling is the same as for the main `configuration.toml`.  BTW this gave me a little trouble because I realised that the special case of undecoded keys (like typos in key names, or maybe obsoleted keys) are not considered an error by `configuration.toml` decoding when I believe they should be.  Fixing this might not be trivial though as it might break existing configs.